python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -130,6 +136,67 @@ def get_asset_key_by_output_name(self, output_name: str) -> AssetKey:
         """Get a dbt node's default asset key by its Dagster output name."""
         return self.node_info_to_asset_key(self.get_node_info_by_output_name(output_name))
 
+    def get_asset_key_for_dbt_unique_id(self, unique_id: str) -> AssetKey:
+        node_info = self.node_info_by_dbt_unique_id.get(unique_id)
+
+        if not node_info:
+            raise DagsterInvalidInvocationError(
+                f"Could not find a dbt node with unique_id: {unique_id}. A unique ID consists of"
+                " the node type (model, source, seed, etc.), project name, and node name in a"
+                " dot-separated string. For example: model.my_project.my_model\n For more"
+                " information on the unique ID structure:"
+                " https://docs.getdbt.com/reference/artifacts/manifest-json"
+            )
+
+        return self.node_info_to_asset_key(node_info)
+
+    def get_asset_keys_by_output_name_for_source(self, source_name: str) -> Mapping[str, AssetKey]:
```

Comment:
If I understand correctly, this method is here to facilitate a very specific pattern and wouldn't really be useful in other situations? If that's the case, I think we should:
- Make sure that this is explicit in the docstring and include a code example of that pattern
- Consider having it return a `Mapping[str, AssetOut]`

Reply:
"This is a convenience method that makes it easy to define a multi-asset that generates all the tables for a given dbt source".

Reply:
It is more ergonomic for most cases for the method to return an AssetOut, but it would be hard for users to customize other properties on the asset out (i.e. make outputs non-required). For that reason I think we should still return a mapping of str to asset key